filter/common.go

```diff
@@ -28,138 +29,74 @@ func filter(filterName string, items []types.CloudItem, filterType types.FilterC
-					return false
-				}
-				log.Debugf("[%s] exclusive filter applied on item: %s", filterName, name)
-				return true
+func isFilterMatch(filterName string, item types.CloudItem, filterType types.FilterConfigType, filterConfig types.IFilterConfig) bool {
+	name := item.GetName()
+	ignoreLabelFound := utils.IsAnyMatch(item.GetTags(), ctx.IgnoreLabel)
+	if ignoreLabelFound {
+		log.Debugf("[%s] Found ignore label on item: %s, label: %s", filterName, name, ctx.IgnoreLabel)
+		if ctx.IgnoreLabelDisabled {
+			log.Debugf("[%s] Ignore label usage is disabled, continuing to apply filter on item: %s", filterName, name)
+		} else {
+			if filterType.IsInclusive() {
+				log.Debugf("[%s] inclusive filter applied on item: %s", filterName, name)
+				return false
 			}
+			log.Debugf("[%s] exclusive filter applied on item: %s", filterName, name)
+			return true
 		}
-		filtered, applied := applyFilterConfig(filterConfig, filterType, item, filterName, stack.Tags)
-		if applied {
-			return filtered
-		}
+	}
+
+	if filterConfig == nil {
+		return false
+	}
+
+	var filterEntityType types.FilterEntityType
+
+	switch item.GetItem().(type) {
 	case types.Access:
-		accessFilter, _ := getFilterConfigs(filterConfig, filterType)
-		if accessFilter != nil {
-			switch item.GetCloudType() {
-			case types.AWS:
-				return isNameOrOwnerMatch(filterName, item, accessFilter.Aws.Names, accessFilter.Aws.Owners)
-			case types.AZURE:
-				return isNameOrOwnerMatch(filterName, item, accessFilter.Azure.Names, accessFilter.Azure.Owners)
-			case types.GCP:
-				return isNameOrOwnerMatch(filterName, item, accessFilter.Gcp.Names, accessFilter.Gcp.Owners)
-			default:
-				log.Warnf("[%s] Cloud type not supported: %s", filterName, item.GetCloudType())
-			}
-		}
-	case types.Database:
-		database := item.GetItem().(types.Database)
-		name := item.GetName()
-		ignoreLabelFound := utils.IsAnyMatch(database.Tags, ctx.IgnoreLabel)
-		if ignoreLabelFound {
-			log.Debugf("[%s] Found ignore label on item: %s, label: %s", filterName, name, ctx.IgnoreLabel)
-			if ctx.IgnoreLabelDisabled {
-				log.Debugf("[%s] Ignore label usage is disabled, continuing to apply filter on item: %s", filterName, name)
-			} else {
-				if filterType.IsInclusive() {
-					log.Debugf("[%s] inclusive filter applied on item: %s", filterName, name)
-					return false
-				}
-				log.Debugf("[%s] exclusive filter applied on item: %s", filterName, name)
-				return true
-			}
-		}
-		filtered, applied := applyFilterConfig(filterConfig, filterType, item, filterName, database.Tags)
-		if applied {
-			return filtered
+		if filterType.IsInclusive() {
+			filterEntityType = types.IncludeAccess
+		} else {
+			filterEntityType = types.ExcludeAccess
 		}
-	case types.Disk:
-		filtered, applied := applyFilterConfig(filterConfig, filterType, item, filterName, types.Tags{})
-		if applied {
-			return filtered
+	case types.Instance, types.Stack, types.Database, types.Disk:
```

Comment:
`types.Access` should move here.

Reply:
Access is handled differently, as it has IncludeAccess and ExcludeAccess. You can find it in the previous switch case

Reply:
my bad, sorry